omod/src/main/java/org/openmrs/web/dwr/DWRProgramWorkflowService.java

```diff
@@ -85,11 +85,50 @@ public PatientProgramItem getPatientProgram(Integer patientProgramId) {
 		return ret;
 	}
 	
-	public Vector<ListItem> getStatesByWorkflow(String uuid) {
-		log.debug("In getStatesByWorkflow with workflow uuid of " + uuid);
+	/**
+	 * @return the ProgramWorkflow matching the given program and workflowLookup
+	 */
+	public ProgramWorkflow getProgramWorkflow(Program program, String workflowLookup) {
+		ProgramWorkflow wf = program.getWorkflowByName(workflowLookup);
+		
+		if (wf == null) {
+			for (ProgramWorkflow programWorkflow : program.getAllWorkflows()) {
+				if (workflowLookup.equalsIgnoreCase(programWorkflow.getConcept().getName().toString())) {
+					wf = programWorkflow;					
+				}
+				else if (workflowLookup.equalsIgnoreCase(programWorkflow.getUuid())) {
+						wf = programWorkflow;						
+				}
+				else if (workflowLookup.equalsIgnoreCase(programWorkflow.getId().toString())) {
+						wf = programWorkflow;
+				}
+			}
+		}
+		return wf;
+	}
+	
+	/**
+	 * @return the ProgramWorkflow matching the given workflowLookup
+	 */
+	public ProgramWorkflow getProgramWorkflow(String workflowLookup) {
+		ProgramWorkflow workflow = Context.getProgramWorkflowService().getWorkflowByUuid(workflowLookup);
+		if (workflow == null) {
+			//TODO Upgrade module to depend on OpenMRS PlatformVersion 2.3.2 and just use #getWorkflow(Integer workflowId);
+			for (Program program : Context.getProgramWorkflowService().getAllPrograms()) {
+				workflow = getProgramWorkflow(program, workflowLookup);
+				if (workflow != null) {
+					return workflow;
+				}
+			}
+		}
+		return workflow;
+	}
+	
+	public Vector<ListItem> getStatesByWorkflow(String workflowLookup) {
+		log.debug("In getStatesByWorkflow with workflow uuid of " + workflowLookup);
 		Vector<ListItem> ret = new Vector<ListItem>();
 		
-		ProgramWorkflow workflow = Context.getProgramWorkflowService().getWorkflowByUuid(uuid);
```

Comment:
What is the value of changing this from the service method call?

Reply:
[This](https://github.com/openmrs/openmrs-core/blob/1.9.11/web/src/main/java/org/openmrs/web/dwr/DWRProgramWorkflowService.java#L86) is how the method was written in 1.9.x , it was expecting an Integer and that's how modules have been calling it. See example [here](https://github.com/openmrs/openmrs-module-reportingcompatibility/blob/635fe83e0ca793b1e0a3c0e7d3923f84c9e10b91/omod/src/main/webapp/analysis/cohortBuilder.jsp#L398).
Somehow, [it was changed](https://github.com/openmrs/openmrs-module-legacyui/blob/master/omod/src/main/java/org/openmrs/web/dwr/DWRProgramWorkflowService.java#L88) to now expect a String (UUID) and that means all modules that are passing an Integer are now failing. 
So, I'm replacing the service call (Context.getProgramWorkflowService().getWorkflowByUuid(uuid);) with a more generic method to make sure modules that use other parameter types (names, and Ids) don't fail.